File: 0217-contains-duplicate/0217-contains-duplicate.cs
public class Solution {
    public bool ContainsDuplicate(int[] nums) {
        HashSet<int> uniqueElements = new HashSet<int>();
        
        foreach (int num in nums)
        {
            if (uniqueElements.Contains(num))
                return true;
            
            uniqueElements.Add(num);
        }     
        return false; 
    }
}

File: 0217-contains-duplicate/0217-contains-duplicate.py
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        nums_set = set()

        for num in nums:
            nums_set.add(num)

        return len(nums) != len(nums_set)

    def containsDuplicate2(self, nums: List[int]) -> bool:
        nums_set = set()

        for num in nums:
            if num in nums_set:
                return True
            nums_set.add(num)

        return False

    